engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -152,6 +159,21 @@
     private DisplayResolutionDependentFBOs displayResolutionDependentFBOs;
     private ShadowMapResolutionDependentFBOs shadowMapResolutionDependentFBOs;
 
+    private static RenderGraph renderGraph = new RenderGraph(); // TODO: Try making this non-static
+
+    // Required for the reflection magic that Systems use.
```

Comment:
On point 1, please create a one-time-only method to pass context and buffer pool immediately after first instantiation of the renderer. Further invocations of the method should do nothing and log a warning.